src/droplet.F

```diff
@@ -550,64 +554,110 @@ subroutine droplet_driver(dt,mtime,xh,uh,ruh,xf,yh,vh,rvh,yf,zh,mh,rmh,zf,mf,zs,
 
      allocate(holes_ind(numHoles))
 
-     !NEW !$acc data create(Depart_ind,holes_ind)
-     !NEW !$acc parallel loop gang vector default(present)
+     i=0
+
      do np = 1, numHoles
        holes_ind(np) = undefined_index
      end do
 
-     i=1
-     !NEW !$acc parallel loop gang vector default(present)
      do np = 1, nparcelsLocal
        if ( (pdata_locind(np,1) .eq. undefined_index) .and. &
             (pdata_locind(np,2) .eq. undefined_index) ) then
-          !NEW !$acc atomic capture
-          holes_ind(i) = np
           i = i + 1
-          !NEW !$acc end atomic
+          holes_ind(i) = np
        end if
      end do
 
      if(timestats.ge.1) time_dropC1=time_dropC1+mytime()
 
-#ifdef MPI
      !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
      ! Step 2: Setup the indirect addresses for the droplets that are departing !
      !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 
-    call setupIndexPointers(Depart,ptrDepart)
+     ! On the CPU, assign pointers (ptrDepart) into indirect address 
+     ! array (Depart_ind) for deprting droplets
+     call setupIndexPointers(Depart,ptrDepart)
 
-    call setupDepartDroplet(Depart_ind,ptrDepart,pdata_neighbor)
+     ! On the CPU, assign the indirect address array (Depart_ind)
+     ! for departing droplets
+     call setupDepartDroplet(Depart_ind,ptrDepart,pdata_neighbor)
 
-    if(timestats.ge.1) time_dropC2=time_dropC2+mytime()
+     if(timestats.ge.1) time_dropC2=time_dropC2+mytime()
 
+#ifdef MPI
     !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
     ! Step 3: Exchange the number of droplets that will enter/leave !
     !         the current MPI region with its nearest neighbor      !
     !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
 
-    call comm_droplet_number(numHoles,Depart,Arrive,pdata, &
-                             pdata_locind,pdata_neighbor)
+    ! On the CPU determine the number of arriving droplet
+    call comm_droplet_number(numHoles,Depart,Arrive)
 
+    
+    ! On the CPU assigns pointers (ptrArrive) into indirect address 
+    ! array (holes_ind) for arriving droplets
```

Comment:
`holes_ind` does not seem to be used in this subroutine.

Reply:
You are correct that the holes_ind is not used in this subroutine, but the ptrArrive is needed in subsequent subroutines to index into holes_ind.

Reply:
Thanks for the clarification.